include/xtensor-io/xblosc.hpp

```diff
@@ -0,0 +1,193 @@
+    {
+        std::ofstream stream(filename, std::ofstream::binary);
+        if (!stream.is_open())
+        {
+            std::runtime_error("IO Error: failed to open file");
+        }
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+    }
+
+    /**
+     * Save xexpression to blosc format in a string
+     *
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline std::string dump_blosc(const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        std::stringstream stream;
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+        return stream.str();
+    }
+
+    /**
+     * Loads a blosc file
+     *
+     * @param stream An input stream from which to load the file
+     * @tparam T select the type of the blosc file
+     * @tparam L select layout_type::column_major if you stored data in
+     *           Fortran format
+     * @return xarray with contents from blosc file
+     */
+    template <typename T, layout_type L = layout_type::dynamic>
+    inline auto load_blosc(std::istream& stream)
+    {
+        std::size_t uncompressed_size;
+        T* uncompressed_buffer = reinterpret_cast<T*>(detail::load_blosc_file(stream, uncompressed_size));
+        std::size_t size = uncompressed_size / sizeof(T);
+        std::vector<std::size_t> shape = {size};
+        auto array = adapt(uncompressed_buffer, size, no_ownership(), shape);
+        return array;
+    }
+    /**
+     * Loads a blosc file
+     *
+     * @param filename The filename or path to the file
+     * @tparam T select the type of the blosc file
+     * @tparam L select layout_type::column_major if you stored data in
+     *           Fortran format
+     * @return xarray with contents from blosc file
+     */
+    template <typename T, layout_type L = layout_type::dynamic>
+    inline auto load_blosc(const std::string& filename)
+    {
+        std::ifstream stream(filename, std::ifstream::binary);
+        if (!stream.is_open())
+        {
+            std::runtime_error("load_blosc: failed to open file " + filename);
+        }
+        return load_blosc<T, L>(stream);
+    }
+
+    struct blosc_config
+    {
+        int clevel;
+        int doshuffle;
+    };
+
+    class xblosc
+    {
+    public:
+        xblosc(int clevel=5, int doshuffle=1): m_clevel(clevel), m_doshuffle(doshuffle) {};
+
+        void configure(blosc_config& config)
+        {
+            m_clevel = config.clevel;
+            m_doshuffle = config.doshuffle;
+        }
+
+        template <class EC, class I>
+        void load(I& input_handler, xarray<EC>& a)
+        {
+            a = load_blosc<EC>(input_handler);
+        }
+
+        template <class E, class O>
+        void dump(O& output_handler, xexpression<E>& e)
+        {
+            dump_blosc(output_handler, e, m_clevel, m_doshuffle);
+        }
+
+    private:
+        int m_clevel;
+        int m_doshuffle;
```

Comment:
Since you've created a blosc_config structure, why not storing it directly instead of replicating it here? (if you implement that change, the constructor that accepts int parameters should forward to that accepting a blosc_config parameter)